src/main/java/de/tum/cit/aet/artemis/programming/service/UserSshPublicKeyService.java

```diff
@@ -0,0 +1,141 @@
+     * the label is extracted from the public key or defaults to a predefined value.
+     *
+     * @param newSshPublicKey the {@link UserSshPublicKey} for which the label is being set.
+     * @param label           the label to assign to the SSH key, or null/empty to use the default logic.
+     */
+    public void setLabelForKey(UserSshPublicKey newSshPublicKey, String label) {
+        if (label == null || label.isEmpty()) {
+            String[] parts = newSshPublicKey.getPublicKey().split("\\s+");
+            if (parts.length >= 3) {
+                String labelFromParts = String.join(" ", Arrays.copyOfRange(parts, 2, parts.length));
+                newSshPublicKey.setLabel(labelFromParts);
+            }
+            else {
+                newSshPublicKey.setLabel(KEY_DEFAULT_LABEL);
+            }
+        }
+        else {
+            newSshPublicKey.setLabel(label);
+        }
+    }
+
+    /**
+     * Retrieves the SSH public key for the specified user by key ID.
+     *
+     * @param user  the {@link User} to whom the SSH key belongs.
+     * @param keyId the ID of the SSH key.
+     * @return the {@link UserSshPublicKey} if found and belongs to the user.
+     * @throws EntityNotFoundException if the key does not belong to the user.
+     */
+    public UserSshPublicKey getSshKeyForUser(User user, Long keyId) {
+        var userSshPublicKey = userSshPublicKeyRepository.findByIdElseThrow(keyId);
+        if (Objects.equals(userSshPublicKey.getUserId(), user.getId())) {
+            return userSshPublicKey;
+        }
+        else {
+            throw new AccessForbiddenException("SSH key", keyId);
+        }
+    }
+
+    /**
+     * Retrieves all SSH public keys associated with the specified user.
+     *
+     * @param user the {@link User} whose SSH keys are to be retrieved.
+     * @return a list of {@link UserSshPublicKey} objects for the user.
+     */
+    public List<UserSshPublicKey> getAllSshKeysForUser(User user) {
+        return userSshPublicKeyRepository.findAllByUserId(user.getId());
+    }
+
+    /**
+     * Deletes the specified SSH public key for the given user ID.
+     *
+     * @param userId the ID of the user.
+     * @param keyId  the ID of the SSH key to delete.
+     * @throws AccessForbiddenException if the key does not belong to the user.
+     */
+    public void deleteUserSshPublicKey(Long userId, Long keyId) {
+        if (userSshPublicKeyRepository.existsByIdAndUserId(keyId, userId)) {
+            userSshPublicKeyRepository.deleteById(keyId);
+        }
+        else {
+            throw new AccessForbiddenException("SSH key", keyId);
+        }
+    }
+
+    /**
+     * Returns whether the user of the specified id has stored SSH keys
+     *
+     * @param userId the ID of the user.
+     * @return true if the user has SSH keys, false if not
+     * @throws AccessForbiddenException if the key does not belong to the user.
+     */
+    public Boolean hasUserSSHkeys(Long userId) {
```

Comment:
I agree with CodeRabbit here. There's nothing stopping us from using `boolean` here, right?

Reply:
I changed it :+1: